AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -179,6 +179,88 @@ public Task<ConcurrentDictionary<string, string>> GetAccountDetailsAsync(string
             return taskSource.Task;
         }
 
+
+        // prevents interference between concurrent calls
+        private readonly SemaphoreSlim _acctSnapLock = new SemaphoreSlim(1, 1);
+
+        /// <summary>
+        /// Get an account detail snapshot
+        /// </summary>
+        /// <param name="accountId">The account Id</param>
+        /// <returns>The account values</returns>
+        public Task<ConcurrentDictionary<string, string>> GetAccountDetailsSnapshotAsync(string accountId)
+        {
+            // Default 5s timeout, you can also change it to a configuration item
+            var cts = new CancellationTokenSource(5000);
+            return GetAccountDetailsSnapshotAsync(accountId, cts.Token);
+        }
+
+        /// <summary>
+        /// Get an account detail snapshot
+        /// </summary>
+        /// <param name="accountId">The account Id</param>
+        /// <param name="cancellationToken">The cancellation token used to cancel the request</param>
+        /// <returns>The account values</returns>
+        public async Task<ConcurrentDictionary<string, string>> GetAccountDetailsSnapshotAsync(
+            string accountId, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(accountId))
+                throw new ArgumentException(nameof(accountId));
+
+            // Ensure connection (your class already has EnsureConnectedAsync)
+            await EnsureConnectedAsync();
+
+            // Prevent concurrent snapshots from interfering with each other/clearing cache
+            await _acctSnapLock.WaitAsync(cancellationToken);
+            try
+            {
+                // 1) First cancel old subscription (ignore exceptions when not subscribed)
+                try { this.clientSocket.ReqAccountDetails(false, accountId); } catch { }
+
+                // 2) Clear cache to ensure this is the current data
+                this.accountUpdates.Clear();
```

Comment:
Make a separate private variable to handle this function to keep it separate from `GetAccountDetailsAsync`